Add fritzbox_callforwarding

jmtatsch opened the pull request:
## Description:
a new component to read out and switch on/off the call forwardings of an AVM Fritzbox

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2358

## Example entry for `configuration.yaml` (if applicable):
```yaml
switch:
  platform: fritzbox_callforwarding
  password: YOUR_PASSWORD
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

jmtatsch committed:
Add fritzbox_callforwarding

jmtatsch committed:
Add fritzbox_callforwarding

homeassistant commented:
Hi @jmtatsch, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6861).

Once you do that we will be able to review and accept this pull request.

Thanks!

jmtatsch committed:
Update .coveragerc

to omit fritzbox_callforwarding from coverage checks

jmtatsch committed:
Update .coveragerc

to omit fritzbox_callforwarding from coverage checks

jmtatsch committed:
Update requirements_all.txt

to include new component fritzbox_callmonitor

jmtatsch committed:
Update requirements_all.txt

to include new component fritzbox_callmonitor

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+        """Enable the call forwarding for a uid."""
+        kwargs = {'NewDeflectionId': self.uid, 'NewEnable': 1}
+        self.fritz_box.call_action(
+            'X_AVM-DE_OnTel:1', 'SetDeflectionEnable', **kwargs)
```
This is very protocol specific. Can this be added to `FritzConnection` dependency?

jmtatsch replied:
Sure, I will check with the repo owner and do it like callmonitor.

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+
+    call_forwardings = []
+    deflections = fritz_box.call_action('X_AVM-DE_OnTel:1', 'GetDeflections')
+    deflection_list = ET.fromstring(deflections['NewDeflectionList'])
```
This is definitely too much protocol specific info. This needs to be moved to a Python package, either FritzConnection or another one.

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+    return call_forwardings
+
+
+def get_call_forwarding_by_uid(fritz_box, uid):
```
Same

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the fritzbox connection."""
+    # pylint: disable=import-error
```
This shouldn't happen. Please remove this comment.

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+    if fritz_box is None:
+        _LOGGER.error('Failed to establish connection to FRITZ!Box '
+                      'with IP: %s', host)
+        return 1
```
1?

balloob commented on homeassistant/components/switch/fritzbox_callforwarding.py:
```diff
@@ -0,0 +1,180 @@
+                      'with IP: %s', host)
+        return 1
+    else:
+        _LOGGER.info('Successfully connected to FRITZ!Box')
```
Please don't log with level info. Use debug instead.

balloob commented on requirements_all.txt:
```diff
@@ -171,6 +171,7 @@ freesms==0.1.1
 
 # homeassistant.components.device_tracker.fritz
 # homeassistant.components.sensor.fritzbox_callmonitor
+# homeassistant.components.sensor.fritzbox_callforwarding
```
How come this is "sensor" ? Please don't update this file manually but follow the instructions in the PR.

jmtatsch replied:
Actually I tried running the script before. It made a lot of changes, decremented a lot of deps etc. 
 # homeassistant.components.device_tracker.fritz
-# homeassistant.components.sensor.fritzbox_callmonitor
-# homeassistant.components.sensor.fritzbox_callforwarding
 # homeassistant.components.sensor.fritzbox_netmonitor
-# fritzconnection==0.6.3
+# fritzconnection==0.6

It seems to have picked up and evaluated an old version of hass so I attempted to alter the file manually.

balloob reviewed (changes requested):
There are too many Fritzbox platforms. Fritzbox will have to be turned into a component that manages the authentication/connection. That way users will only have to add their username and password to the configuration once.

This PR also includes a lot of protocol specific information that cannot be included in Home Assistant. Please see if you can contribute it back to FritzConnection. If the author is not open for that, create a new dependency.

craigjmidwinter committed:
use change light level to avoid variable ramp speeds (#6860)

balloob committed:
Handle aiohttp task cancellation better (#6862)

tchellomello committed:
Introduced Ring binary sensors and refactored Ring component (#6520)

* - Introduced Ring binary_sensor.

- Added unittest for Ring binary_sensor.

- Bumped ring_doorbell 3rd party module.

* Updated requirements

* Added correct file for unittest

* - Introduced Ring binary_sensor.

- Added unittest for Ring binary_sensor.

- Bumped ring_doorbell 3rd party module.

* Updated requirements

* Added correct file for unittest

* Added extra sensors last_ding and last_motion

* Modified Ring binary_sensor and sensor to inherit DOMAIN configuration

* Moved static to top ring.py

* Fixed requirements

* Bump version ring_doorbell to 0.1.2

* testing unittests

* Use hass.data dict instead GLOBALS

* Fixed unittests

* Bump ring_doorbell to 0.1.3

* Updated unittest and coverted to use decorator @requests_mock.Mocker()

* Updated ring_session with lower case

fabaff committed:
Upgrade sendgrid to 3.6.5 (#6866)

fabaff committed:
Upgrade sphinx-autodoc-typehints to 1.2.0 (#6865)

jacobtomlinson committed:
Added Met Office weather and sensor components (#6742)

* Added Met Office weather and sensor components

* Removed unnecessary dependancy

* Generated requirements

* Fix time interval

* Updated coverage

* Some review changes

* Allow user to specify lat and lon in component

* Added missing import

* Fixed unit

* Fixed import indent

* Updated condition to use CONDITION_CLASSES

fabaff committed:
Upgrade speedtest-cli to 1.0.3 (#6867)

JesseWebDotCom committed:
Exposed more attributes, enabled play_media tv show or season episodes

JesseWebDotCom committed:
mend

Exposed more attributes, enabled play_media tv show or season episodes

JesseWebDotCom committed:
Fixed setting username and content rating data

tchellomello committed:
Bumped amcrest module to 1.1.5 (#6872)

fabaff committed:
Upgrade pytz to 2017.02 (#6875)

fabaff committed:
Upgrade aiohttp_cors to 0.5.2 (#6874)

fabaff committed:
Upgrade sqlalchemy to 1.1.8 (#6873)

jmtatsch committed:
Update fritzbox_callforwarding.py

fixed doc string, raise a proper exception ...

jmtatsch committed:
Update fritzbox_callforwarding.py

fixed doc string, raise a proper exception ...

jmtatsch committed:
Update fritzbox_callforwarding.py

indentation and doctoring fixes

jmtatsch committed:
Update fritzbox_callforwarding.py

indentation and doctoring fixes

jmtatsch committed:
Update requirements_all.txt

jmtatsch committed:
Update requirements_all.txt

ChristianKuehnel committed:
added  support for Fibaro FGR-222 (similar to FGRM-222) (#6890)

Danielhiversen committed:
Fluxled (#6892)

* Update flux_led lib

JesseWebDotCom committed:
Fixed log errors if session username or content rating is blank

Commit:
Fix Tado climate set off mode (#6848)

especially-relative committed:
Fox UMP volume set (#6904)

async needs consistant paramater namming accross platforms. This fixes
UMP's volume set method by renaming the paramater.

balloob committed:
Move examples out (#6908)

* Remove examples from main repo

* Simplify README

* Point screenshot for components at dev branch for now

balloob committed:
Update README.rst

tchellomello committed:
Makes amcrest.sensor to handle properly the scan_interval option. (#6885)

* Makes amcrest.sensor to handle scan_interval option as expected.

* Added _LOGGER.debug statement for troubleshooting.

* Fixed lint

tchellomello committed:
Make sensor.ring to handle scan_interval option as expected. (#6886)

* Make sensor.ring to handle scan_interval option as expected.

* Fix lint

* Fixed lint timedelta

nugget committed:
Eliminate needless async_add_job invocation of async_add_devices (#6864)

especially-relative committed:
Onkyo update (#6906)

* Update onkyo req, change volume to int

* Update Onkyo

Updates onkyo component. Pulls added sources (Bluetooth, built-in
streaming, etc.)

* Regenerated requirements_all.txt via script

* Update onkyo.py

* Update requirements_all.txt

turbokongen committed:
Fix for #6691 Neato Connection error handling (#6731)

* Responsiveness

* Delay was not needed as commands does not return until done.

* Offline error catch

* Remove unneeded code

gurumitts committed:
Adds support for the PlugInDimmer hardware (#6915)

emlove committed:
Support for zwave light transitions (#6868)

* Support for zwave light transitions

* Dimming duration is optional

* Updated supported_features to show transition

mezz64 committed:
Bump pyHik library version to support more cameras (#6921)

pavoni committed:
Update vera cover refresh logic (#6897)

* Update cover update logic to fix compatibility bug. Bump vera library.

* Tidy.

* Add missed file.

balloob committed:
Update frontend

balloob committed:
Automation: initial state > restore state (#6911)

* Automation: initial state > restore state

* Clean up code

* Ensure MQTT defaults are used.

* Ensure failed platforms always return None

* Automation: write state to state machine after start

Danielhiversen committed:
Upgrade flux_led to 0.17 (#6929)

* Update flux_led lib

fabaff committed:
Upgrade paho-mqtt to 1.2.1 (#6928)

fabaff committed:
Upgrade distro to 1.0.4 (#6926)

fabaff committed:
Upgrade Sphinx to 1.5.4 (#6927)

KlaasH committed:
Allow token authentication for 'hook' switch component (#6922)

The Hook switch component currently uses username/password to get a token
from the Hook web app to use for subsequent requests. This adds an option
to specify the token directly in config.

Makes the 'password' and 'token' options mutually exclusive since
otherwise one would have to take precedence, and it seems worth
preventing people from filling their config with passwords that don't
even need to be there.

pavoni committed:
WIP - Fix bug in state handling in Vera Switch and Light (#6931)

* Fix bug in state handling.

* Tidy.

craigjmidwinter committed:
total connect alarm support (#6887)

* total connect alarm support

* linting fixes

* linting fixes

* docstring

* docstring

* use sync and update coveragerc

* remove unused import

* changes as per notes

* Update HA code style

balloob committed:
Initial state over restore state (#6924)

* Input Boolean: initial state > restore state

* Input select: initial state overrules restored state

* Input slider: initial state overrule restore state

* Lint

* Lint

citruz committed:
Eddystone Beacon Temperature Sensor (#6789)

* Added eddystone_temperature platform.

* Fixed style issues.

* Fixed style issues #2.

* Fixed style issues #3.

* Added new platform to .coveragerc

* Refactored platform to use the beacontools package.

* Fixed style issues and added beacontools to excluded requirements.

* Removed obsolete constants and added pylint exception.

* Added blank line

* Updated beacontools to version 1.0.0

* Updated beacontools to version 1.0.1

* Forgot to regenerate requirements_all

* Minor changes

JesseWebDotCom committed:
String formatting, import order, type check fixes

pvizeli committed:
Add android ip webcam support for aiohttp2 (#6940)

pavoni committed:
Bump pywemo version. Fixes Osram/Sylvania Lightify tunable white bulbs. (#6946)

Add an optional extended description…

tchellomello committed:
Clean artifacts after running Ring tests. (#6944)

* Clean artifacts after running Ring tests.

* Clean artifacts from top level test_ring.py

emlove committed:
Rename zwave nodes by node ID instead of entity ID (#6938)

jawilson committed:
Report proper features in mqtt_json light (#6941)

* Add tests for supported features in mqtt_json (it fails)

* Fix supported features in mqtt_json

titilambert committed:
Add multi phone numbers support (#6605)

* Add multi phone numbers support

* Update fido.py

fabaff committed:
Upgrade Sphinx to 1.5.5 (#6947)

fabaff committed:
Upgrade py-cpuinfo to 3.0.0 (#6948)

balloob committed:
Fix automations listening to HOMEASSISTANT_START (#6936)

* Fire EVENT_HOMEASSISTANT_START automations off right away while starting

* Actually have core state be set to 'starting' during boot

* Fix correct start implementation

* Test and deprecate event automation platform on start

* Fix doc strings

* Remove shutting down exception

* More strict when to mark an instance as finished

* Add automation platform to listen for start/shutdown

* When we stop we should wait till it's all done

* Fix testing

* Fix async bugs in tests

* Only set UVLOOP when hass starts from CLI

* This hangs normal asyncio event loop

* Clean up Z-Wave node entity test

pvizeli committed:
Fix startup of sonos / snapshot handling / error handling (#6945)

* Fix startup of sonos / snapshot handling / error handling

* Use decorator for coordinator relay

* fix lint

* Fix unittest

* Move subscribe into executor

MartinHjelmare committed:
Upgrade mysensors dep and callbacks (#6950)

diogos88 committed:
Added average temperature for the day before and the current period (#6883)

* Added average temperature for the day before and the current period

* Fixed "line too long" warnings

* Fixed "indentation contains tabs" and "indentation contains mixed spaces and tabs" warnings

* Fixed "trailing whitespace" warnings

* upgrade pyhydroquebec requirements to version 1.1.0

fabaff committed:
Upgrade sqlalchemy to 1.1.9 (#6955)

emlove committed:
Update kodi for aiohttp2 (#6967)

aufano committed:
Fix current_temperature is rounded (#6960)

* Fix current_temperature is rounded

* fix  Unnecessary parens after 'if'

JesseWebDotCom committed:
import order fix

pvizeli committed:
Bugfix time and task coro (#6968)

* Bugfix time and task coro

* fix also other create_task

* fix tests

* fix lint in test

pvizeli committed:
Initial import for HassIO (#6935)

* Initial import for HassIO

* Cleanup api code for views

* First unittest for view

* Add test for edit view

* Finish unittest

* fix addons test

* cleanup service.yaml

* Address first round with ping command

* handle timeout dynamic

* fix lint

andrey-git committed:
Preserve customize glob order. (#6963)

* Preserve customize glob order.

* add tests

viswa-swami committed:
Foscam Camera: Adding exception handling when fetching the camera image to avoid python exception errors when host is not reachable or rather any url error to camera (#6964)

* Adding exception handling when fetching the camera image to avoid python errors when host is not reachable or any url errors to camera

* Added exception as ConnectionError instead of plain except

* Added exception as ConnectionError instead of plain except. Removed the unused error handle

rytilahti committed:
light.yeelight: catch i/o related exceptions from the backend lib (#6952)

Fixes/mitigates problems with #5949 and #6624

happyleavesaoc committed:
Crime Reports sensor (#6966)

* add crimereports

* add crimereports metadata

* implicit interval

* remove zone support

BillyNate committed:
From Dusk till Dawn (#6857)

* Added dawn, dusk, noon and midnight to the Sun component

* Created a helper method for the solar events

emlove committed:
Tests for zwave services (#6937)

* Initial tests for zwave services

* Fix linter issues

* Complete zwave service tests

balloob committed:
Fix control+c quitting HASS (#6974)

mezz64 committed:
Update Emby for aiohttp v2 (#6981)

rytilahti committed:
switch.tplink: upgrade to the newest upstream release which adds support for plugs using the newer communication protocol (#6979)

rytilahti committed:
switch.tplink: bump pyhs100 version requirement (#6986)

emlove committed:
Add tests for ZWaveDeviceEntityValues helper (#6978)

* Add tests for ZWaveDeviceEntityValues helper

* Add remaining coverage

tchellomello committed:
Bump Amcrest module to 1.1.8 (#6990)

Fixed traceback when calculating SD card percent storage

   self._state = self._camera.percent(sd_used[0], sd_total[0])
AttributeError: 'Http' object has no attribute 'percent'

happyleavesaoc committed:
update gstreamer (#6987)

balloob committed:
Warn if start takes a long time. (#6975)

* Warn if start takes a long time.

* ps - cleanup

* Tweak message

* Add tests

* Tweak messagE

balloob committed:
Upgrade to aiohttp 2.0.6 (#6992)

balloob committed:
Make discovery not block start (#6991)

* Make discovery not block start

* Fix tests

balloob committed:
Downgrade aiohttp to 205 (#6994)

pavoni committed:
Bump pywemo version. (#7004)

balloob committed:
Fix Synology camera content type (#7010)

balloob committed:
Fix two more instances of JSON parsing synology (#7014)

Add an optional extended description…

Commit:
Bump pyalarmdotcom to support new version of aiohttp (#7021)

Add an optional extended description…

fabaff committed:
Fix US states check (fixes #7015) (#7017)

balloob committed:
Remove deprecated remote classes (#7011)

* Remove deprecated remote classes

* Lint

* Fix tests

* Lint

fabaff committed:
Replace 'vendor_id' with 'arch' (fixes #7003) (#7023)

micw committed:
more tests for slugify (#7027)

emlove committed:
Additional ZWave coverage (#7024)

* Additional ZWave coverage

* setup_platform tests

happyleavesaoc committed:
bump ups version (#7033)

happyleavesaoc committed:
update usps version (#7035)

happyleavesaoc committed:
update fedex (#7034)

pezinek committed:
Google TTS can't read percent sign (#6971) (#7030)

micw committed:
Feature/min max improvements (#6786)

* Fix #6783, remove a test that makes no sense anymore

* Fix #6784

* Fix typo in docstring

* Fix handling of known->unknown state, extended test, fix lint errors

* Better handling of mismatch in unit of measurement.

Set state to "unkown" and unit of measurement to "ERR" if unit of measurement differs between aggregatet states.
Add entity_id to logged warning.

* Make icon configurable

* Fix typo

* Fix lint

* Fix lint

* Fix lint

* Add option to set entity_id on min_max sensor

* Fix lint logging-not-lazy

* Revert "Add option to set entity_id on min_max sensor"

This reverts commit 4685f266477dfaba92cf8a256ecfe62c929c877b.

* Revert "Make icon configurable"

This reverts commit fe45aec82d9a07e730bebb4d47dc9618fb695fe6.

* Fixes

fabaff committed:
Upgrade psutil to 5.2.2 (#7037)

fabaff committed:
Upgrade sendgrid to 4.0.0 (#7038)

DavidMStraub committed:
Missing line name restriction added (fixes #7039) (#7040)

amelchio committed:
Plug file leak on LIFX unregister (#7031)

* Plug file leak on LIFX unregister

The aiolifx 0.4.4 release closes its socket when the unregister callback is
called. This plugs a file descriptor leak but also means that we must be
careful to not use the device after it goes unavailable.

Also, when a light reappears, it has a new device that must be used.

* Do not test self.available in service calls

The core will learn to handle that.

pvizeli committed:
Make core to look avilable state of device on servicecall (#7045)

fabaff committed:
Remove configuration sample (#7048)

pvizeli committed:
Bugfix wait on start event (#7013)

* Bugfix wait on start event

* Paulus sugestion

* Change handling with stop_track_task

* Add new unittests

* Update test_core.py

pvizeli committed:
Bugfix slider (#7047)

* Bugfix slider

* Update input_slider.py

* Update input_slider.py

Commit:
Add MaryTTS platform (#6988)

* Add MaryTTS platform

* Fix lint error

* Doc link, config and formatting fixes

* Remove stuff not needed with aiohttp2

* Get rid of unnecessary else statement

MartinHjelmare committed:
Fix mysensors callback (#7057)

* Fix mysensors callback

* All messages was not triggering proper updates. Fix by checking all
  child value types each update.

* Upgrade mysensors dep

* Fix pickle persistence when upgrading.

balloob committed:
Constrain chardet to 2.3 (#7063)

micw committed:
Fix/slugify with german umlaut ss (#7029)

* more tests for slugify

* Fix german umlauts in slugify

* Update __init__.py

balloob committed:
Speed up aiohttp (#7064)

balloob committed:
Upgrade netdisco to 1.0.0rc2 (#7008)

* Upgrade netdisco to 1.0.0rc2

* fix tests

sander76 committed:
Telegram bot component (incl. webhook and polling platform) (#6913)

* first commit.

* removed pointless string statement

* manually removed  # homeassistant.components.telegram_webhooks from requirements_all.txt

* deleted obsolete file.

* coveragerc abc

fabaff committed:
Upgrade paho-mqtt to 1.2.2 (#7066)

pvizeli committed:
Fix handling with register callbacks on added_to_hass (#7067)

pvizeli committed:
Lutron. Bugfix callback registration. (#7042)

* Lutron. Bugfix callback registration.

* Change handling to event

hawk259 committed:
Adding AlarmDecoder platform (#6900)

* Added AlarmDecoder platform

* remove try/catch for generic execption

* Changes for @pvizeli, thanks for the review!

Removed _ prefix from normal function variables
Removed _hass as it will be set via .hass for us
Broke out the three config (socket, serial, usb) and use vol.Any
Added support for USB I think, don't have device, but should work
Removed components dictionary, was form old group all code that didn't work

* Fix hass string handling

turbokongen committed:
Add communication data attributes to Zwave node_entity (#6813)

* Add quality attribute to node

* Move quality to node_entity

* adjustments

* Line lenght

* flake8

* Cleanup and add each entry from getNodeStatistics as attribute

* resolve conflict

* Move NETWORK to hass.data

* Test Done

* Cleanup from review

* Resolve conflicts

* Review changes

* Long lines....

* blank line

* import-error-disable

* Update tests Part1... Again

* Hound

* Argh!

* Argh!

* YABBADABBADOOOOOOO!

* Enhance tests

* hound

* Resolve

* resolve tests...

andrey-git committed:
Add product_name attribute to zwave nodes. (#7071)

robbiet480 committed:
Merge pull request #6869 from JesseWebDotCom/dev

Exposed more attributes, enabled play_media tv show or season episodes

robbiet480 committed:
Bump braviarc version to 0.3.7 (#7078)

titilambert committed:
Fix account balance in fido sensor (#7077)

MrMep committed:
MQTT: Managing binary payloads (#6976)

* Managing binary payloads

Hello,
background: I wrote a HA camera component that gets the image from a binary payload. I'm testing it with Zanzito (https://play.google.com/store/apps/details?id=it.barbaro.zanzito) and it works apparently well: it gets the image and correctly displays it in the front-end.
But I had to make the changes I'm proposing here: the message was being blocked because the utf-8 decoding failed.
As far as I know, the utf-8 encoding is required for the topic, not for the payload. What I did here was try the utf-8 decoding, but even if unsuccessful, it dispatches the message anyway.
Is there anything else I'm missing?
thanks
Gianluca

* Update __init__.py

* Update __init__.py

* Update __init__.py

* git test - ignore

* Should work

* minor fixes

* updated mqtt/services.yaml

* added two tests, modified threaded subscribe

* removing polymer

* requested changes

* requested changes - minor fix

* security wrap around payload_file_path

* services.yaml updated

* removed file publishing

* minor fix

colinodell committed:
Bump qnapstats library version to 0.2.4 (#7085)

turbokongen committed:
Add support fo map data from Neato (#6939)

* Responsiveness

* Delay was not needed as commands does not return until done.

* Add support for cleaning maps and cleaning data

* Hound

* Docstring

* Update requirements

* Review changes

* External lib now returns the raw data.

* debug

* Sensor did not refresh

* Error handling

* Issue warning on connection error

* Update requirements

* Review changes

Commit:
Added initial support for IKEA Tradfri Gateway (#7074)

* Added initial support for IKEA Tradfri Gateway

* Pinned requirement

* Fixed lint-errors

* Added file to .coveragerc

* Trying to fix commit

* Fixed requirements_all again

* Minor reorder of code

* Minor reorder of code

* Made the changes suggested by @balloob

* Made the changes suggested by @balloob and removed debug

* Update tradfri.py

andrey-git committed:
Better thread safety in zwave node_entity (#7087)

* Better thread safety

* Update node_entity.py

robbiet480 committed:
Break product name up into manufacturer name and product name

robbiet480 committed:
Merge pull request #7094 from robbiet480/zwave-manufacturer-product-names

Break Z-Wave product name up into manufacturer name and product name

emlove committed:
Uber version bump (#7100)

emlove committed:
No product ids configured should fetch all ids (#7091)

fabaff committed:
Upgrade aiohttp to 2.0.7 (#7106)

fabaff committed:
Update file header, add const for defaults, and update log messages (#7110)

fabaff committed:
Upgrade speedtest-cli to 1.0.4 (#7105)

fabaff committed:
Remove globally disabled pylint issue and update docstrings (#7111)

jmtatsch committed:
factored out the call forwarding logic to into seperate package

jmtatsch committed:
merge

jmtatsch committed:
bumped the other platforms that depend on fritzconnection an gen requirements

balloob commented:
Looks like your branch got messed up and contains a lot of commits that are not yours. Please clean it up.

balloob commented:
This PR seems to have gone stale. Closing it. You can reopen it when you're ready to finish it.
